File: PropertyManagementSystem/PropertyManagementSystem/Models/Apartment.cs
using System.ComponentModel.DataAnnotations;

namespace PropertyManagementSystem.Models
{
    public class Apartment: BaseModel
    {
        [Required,StringLength(10)]
        public string ApartmentNum { get; set; }

        [Required,StringLength(10)]
        public string Floor { get; set; }        
        public float AreaSqft { get; set; }

        public string AppUserId { get; set; }
        public AppUser Owner { get; set; }

        public long BuildingId { get; set; }
        public Building Building { get; set; }
    }
}

File: PropertyManagementUI/src/app/Models/Apartment.ts
import { BaseModel } from "./BaseModel";
import { AppUserDto } from "./AppUser";
import { Building } from "./Building";

export interface Apartment extends BaseModel {
    ApartmentNum: string | null;
    Floor: string | null; 
    AreaSqft: number | null;
    AppUserId: string | null;
    Owner: AppUserDto | null;
    BuildingId: number | null;
    Building: Building | null;
}